File: python/6_kyu/convert_string_to_camel_case.py
# Convert String to Camel Case
# https://www.codewars.com/kata/517abf86da9663f1d2000003/train/python

import re

# Solution 1: re.split
def to_camel_case(text):
  if "-" in text or "_" in text:
    text = re.split('-|_', text)
  else:
    return ''
  
  ans = [text[0]]

  for i in range(1, len(text)):
    ans.append(text[i][0].upper() + text[i][1:])
  
  return "".join(ans)

# Solution 2: replace
def to_camel_case(text):
  text = text.replace("_", "-").split("-")
  
  if len(text) == 0:
    return ""
  
  return text[0] + "".join([i.capitalize() for i in text[1:]])



print(to_camel_case(''))
print(to_camel_case('the_stealth_warrior'))
print(to_camel_case('The-Stealth-Warrior'))
print(to_camel_case('A-B-C'))

File: javascript/6_kyu/convert_string_to_camel_case.js
// Convert string to camel case
// https://www.codewars.com/kata/517abf86da9663f1d2000003/train/javascript

function toCamelCase(text) {
  text = text.replace(/_/gi, "-").split("-");

  if (text.length == 0) {
    return "";
  }

  var ans = [text[0]];

  for (var i = 1; i < text.length; i++) {
    ans.push(text[i].charAt(0).toUpperCase() + text[i].slice(1));
  }

  return ans.join("");
}

console.log(toCamelCase(""));
console.log(toCamelCase("the_stealth_warrior"));
console.log(toCamelCase("The-Stealth-Warrior"));
console.log(toCamelCase("A-B-C"));
console.log(toCamelCase("The_coding_warrior_and_eye"));